internal/vulnstore/postgres/gc.go

```diff
@@ -0,0 +1,260 @@
+package postgres
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+
+	"github.com/google/uuid"
+	"github.com/jackc/pgtype"
+	"github.com/jackc/pgx/v4"
+	"github.com/jackc/pgx/v4/pgxpool"
+	"golang.org/x/sync/semaphore"
+)
+
+func (s *Store) GC(ctx context.Context, keep int) (int64, error) {
+	const (
+		// GCThrottle sets a limit for the number of deleted update operations
+		// (and subsequent cascade deletes in the uo_vuln table) that can occur in a GC run.
+		GCThrottle = 50
```

Comment:
increased this, I'm a little fearful of a lot of cascade deletes happening at once, they will cause individual row locks on the table - not sure how that plays out in other parts of the app, but feel limiting them is a good idea. not immediately sure what that limit should be.

Reply:
yeah, not obvious. We should figure out a way to A/B test this.

Reply:
Should it stay a constant in that case? Or should this become a config

Reply:
I think we need to figure out a system to observe it first, then we can work in a toggle. This is a purely internal data structure, so I'm not worries.